editor/code/src/goals.ts

```diff
@@ -78,6 +78,10 @@ export class InfoPanel {
   // notify the info panel that we have fresh goals to render
   requestDisplay(goals: GoalAnswer<PpString>) {
     this.postMessage("renderGoals", goals);
+    let vizx = extensions.getExtension("inQWIRE.vizx");
+    if (vizx?.isActive) {
+      commands.executeCommand("vizx.lspRender", goals);
+    }
```

Comment:
Looks good to me as a prototype, will merge so you can continue your testing.

I propose we indeed move towards a registration mechanism like done in most parts of VSCode, so eventually we can replace this code with your extension doing in activation:
```typescript
let coq-lsp = extensions.getExtension("coq-lsp").?API_v1
coq-lsp.registerGoalProcessorString((goals) => lspRender(goals))
```

We may need a bit different API once we need to properly manage the UI resources both extensions handle, but that should be a good start.